deps/rabbit/src/rabbit_queue_type.erl

```diff
@@ -307,6 +307,7 @@ is_compatible(Type, Durable, Exclusive, AutoDelete) ->
 declare(Q0, Node) ->
     Q = rabbit_queue_decorator:set(rabbit_policy:set(Q0)),
     Mod = amqqueue:get_type(Q),
+    ok = check_vhost_queue_limit(Q),
```

Comment:
This will raise an exception if the limit is reached. The error will neither be logged nor propagated to the channel to be converted into a channel exception.

Reply:
Ah, I saw that `rabbit_channel` called the handle_method function from a try-catch so I thought it be OK to throw the error. But, I realize that `rabbit_queue_type:declare` and `rabbit_amqqueue:declare` might actually be called from other sources, so the right thing would be for `rabbit_queue_type:declare` to return a
`{protocol_error, Type :: atom(), Reason :: string(), Args :: term()}` if a limit is met?

Reply:
yes a `protocol_error` should be a ok to return as it is already in the return spec for the declare function.